### PR TITLE
feat(dif): Add WASM support

### PR DESCRIPTION
**Pull request by mitsuhiko**
Extends the protocol and native processing plugin for WASM support.

## Screenshot

image

## Dependencies

* [x] https://github.com/getsentry/symbolic/pull/166: symbolic changes for symbolicator and sentry-cli
* [x] https://github.com/getsentry/symbolicator/pull/301: changes to symbolicator to enable WASM processing
* [x] https://github.com/getsentry/relay/pull/852: updates to relay for the protocol
* [x] https://github.com/getsentry/develop/pull/209: protocol documentation changes
* [x] https://github.com/getsentry/sentry-cli/pull/863: for sentry-cli difutil changes
* [ ] https://github.com/getsentry/sentry-javascript/pull/3080: for the JavaScript SDK

For this to work the symbolic sentry library also needs to be recompiled with the latest symbolic (or a version containing https://github.com/getsentry/symbolic/pull/166).

## Followup Work

* [x] UI does not deal well with the fact that WASM modules are relatively addressed exclusively and have no size.

### CODE REVIEW

File: scripts/ensure-venv.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ if [ -n "$VIRTUAL_ENV" ]; then
 
     # TODO: Update this to strictly check .python-version
     if [ "$SENTRY_PYTHON3" = "1" ]; then
-        python -c "import sys; sys.exit(sys.version_info[:2] != (3, 6))" ||
+        python -c "import sys; sys.exit(sys.version_info[:2] != (3, 8))" ||
```
Comment (BYK) [inline scripts/ensure-venv.sh:30]: Whoa whoah whoah, don't you think we are going too fast Jimmy? 😝

Comment (mitsuhiko) [reply scripts/ensure-venv.sh:30]: hah. accident

File: src/sentry/lang/native/processing.py [hunk 2]
```diff
@@ -283,6 +290,32 @@ def _handles_frame(data, frame):
...
 
 
+def _expand_addr_mode(addr_mode, modules):
+    if addr_mode in (None, "abs"):
+        return "abs"
+    if addr_mode.startswith("rel:"):
+        arg = addr_mode[4:]
+
+        for idx, module in enumerate(modules):
+            if module.get("debug_id") == arg:
```
Comment (jan-auer) [inline src/sentry/lang/native/processing.py:300]: We've seen crashes with 700 modules and 20k+ frames. It's probably not a huge performance gain, but I'd go with a map here.

File: src/sentry/lang/native/processing.py [hunk 3]
```diff
@@ -283,6 +290,32 @@ def _handles_frame(data, frame):
     return is_native_platform(platform) and "instruction_addr" in frame
 
 
+def _expand_addr_mode(addr_mode, modules):
+    if addr_mode in (None, "abs"):
+        return "abs"
+    if addr_mode.startswith("rel:"):
+        arg = addr_mode[4:]
+
+        for idx, module in enumerate(modules):
```
Comment (jan-auer) [inline src/sentry/lang/native/processing.py:299]: Before iterating modules, you can use symbolic to check if this is even a debug id. Alternatively, what if we try the number first, and only then check debug_ids which is more "expensive".

**Review by jan-auer — APPROVED** [review]

**Review by jan-auer — APPROVED** [review]

**Pull request merged**